src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -1014,11 +1014,11 @@ void BgDyYoseizo_ParticleDraw(BgDyYoseizo* this, GlobalContext* globalCtx) {
 
     for (i = 0; i < 200; i++, particle++) {
         if (particle->alive == 1) {
-            if (phi_s3 == 0) {
-                gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(gGreatFairyParticleAppearDL));
+            if (!isMaterialSet) {
+                gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(gGreatFairyParticleMaterialDL));
                 gDPPipeSync(POLY_XLU_DISP++);
 
-                phi_s3++;
+                isMaterialSet++;
```

Comment:
Incrementing a bool?

Reply:
as long as it's not false

this pattern is common to near all draw effect functions, `BossFd_DrawEffects` also has it as an incremented boolean, many are just unnamed or half-used as bools somehow

Reply:
Arguable that should disqualify them from being counted as bools, then... I'm sure C99 would complain.

Reply:
```suggestion
                isMaterialSet += true - false;
```
:uhhuh:

No but, I'd be ok with using `isMaterialSet = 0`, `isMaterialSet == 0` if it's that bad for C to increment "false", but the `isXXX` bool name should stay or what to name it? `isMaterialSetFlag`? `materialFlag`? `materialSet`?

Reply:
i mean this C standard doesnt have bools, so trying to force ideals from the newer standard onto something that wasnt written for it seems weird to me. But in the end it doesnt matter to me too much. I would prefer it be treated as a bool all around cause it clearly is, regardless of how the value gets set

Reply:
I reverted this change to master, as I figured all these actor-effects systems should probably be named/documented at once.